hr_switzerland/models/res_users.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Stephane Eicher <seicher@compassion.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+import logging
+
+from odoo import models, api, _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class ResUser(models.Model):
+    _inherit = 'res.users'
+
+    @api.model
+    def asterisk_connect(self, log=True):
+        for ast_user in self:
+
+            user, ast_server, ast_manager = \
+                self.env['asterisk.server']._connect_to_asterisk()
+
+            channel = '%s/%s' % (ast_user.asterisk_chan_type, user.resource)
+
+            _prefix = '*31' if log else '*32'
+            extension = _prefix + ast_user.internal_number
+
+            try:
+                ast_manager.Originate(
+                    channel,
+                    context='default',
+                    extension=extension,
+                    priority=1,
+                    timeout=unicode(ast_server.wait_time * 1000),
+                    caller_id=ast_user.internal_number,
+                    account=ast_user.cdraccount)
+            except Exception, e:
+                _logger.error(
+                    "Error in the Originate request to Asterisk server %s",
+                    ast_server.ip_address)
+                _logger.error(
+                    "Here are the details of the error: '%s'", unicode(e))
+                raise UserError(
```

Comment:
I'm not sure that we should raise the error? It means the user won't be able to check in or check out in case of error. I think it's okay if the asterisk logon/logoff fails. What do you think? Or maybe here we could use a small toast message if we inherit from this module that I looked for some time : [OCA/web_notify](https://github.com/OCA/web/tree/10.0/web_notify)